osu.Game/Rulesets/UI/FallbackSampleStore.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using osu.Framework.Audio;
+using osu.Framework.Audio.Sample;
+using osu.Framework.Audio.Track;
+using osu.Framework.Bindables;
+
+namespace osu.Game.Rulesets.UI
+{
+    public class FallbackSampleStore : ISampleStore
+    {
+        private readonly ISampleStore primary;
+        private readonly ISampleStore secondary;
+
+        public FallbackSampleStore(ISampleStore primary, ISampleStore secondary)
+        {
+            this.primary = primary;
+            this.secondary = secondary;
+        }
+
+        public void Dispose()
+        {
+            primary.Dispose();
+            secondary.Dispose();
```

Comment:
This is 100% prone to error / dangerous. Probably should be the consumer's responsibility to dispose.

Reply:
oh yeah, i meant to remove this (i already moved handling to the consumption point after noticing this).